src/isort/mod.rs

```diff
@@ -137,6 +144,11 @@ fn annotate_imports<'a>(
                     module: module.as_ref(),
                     names: aliases,
                     level: level.as_ref(),
+                    trailing_comma: if split_on_trailing_comma {
+                        trailing_comma(import, locator)
+                    } else {
+                        TrailingComma::default()
+                    },
```

Comment:
I also tweaked the order of operations: instead of collecting locations, and looking at the locations at the end, we detect a trailing comma for each block, and then propagate it through the code as we merge imports.